### ISSUE TITLE
feat: Using Bootstrap SCSS + Bootstrap-Vue SCSS

### ISSUE THREAD

**Issue by sts-ryan-holton** [bootstrap-vue/bootstrap-vue#2201]
Does Bootstrap Vue come with SCSS? If so, I'm importing my SCSS files and SCSS variables via: https://nuxtjs.org/api/configuration-css into the project but this doesn't seem to be working?

**Comment by tmorehouse** [comment]
Yeah, I have been thinking of adding a PR to generate both an SCSS file and regular css file on build, as there are a few Bootstrap-Vue CSS files that could take advantage of Bootstraps Variables.

**Comment by tmorehouse** [comment]
I've got a few of the files converted to SCSS on my dev box, and just working on the best way to bundle them (i.e. individual files + combined `.scss` file, as well as default `.css` file (for those using bootstrap defaults).

Currently the `es`  files include css imports, but that will need to be filtered out.

**Comment by jackmu95** [comment]
@tmorehouse What about removing the imports form the source JS files completely and create and index `bootstrap-vue.scss` file to include all custom SCSS files from the components.

This way it's possible to compile all custom CSS to one file by using the index file, it's possible to include the index SCSS file in a existing project or use the SCSS files on a component basis.

**Comment by tmorehouse** [comment]
@jackmu95 Yeah, that is another option as well.

I would still like to keep the SCSS files specific to a component in the component directory, but it is possible to use something like `bundle-scss/sass` to merge them into a complete file, and then run the sass compiler on them after in the build script.   Just need to make sure HMR in dev mode triggers reloads on style changes.

**Comment by tmorehouse** [comment]
Although if we use the [`collect-sass`]9https://github.com/nathancahill/rollup-plugin-collect-sass0 rollup plugin, we can create the compiled `css` from the included `scss` files (using default bootstrap settings), but we would still need to use `bundle-sass` to generate a complete SCSS file that people can use (which shouldn't be an issue)

**Comment by jackmu95** [comment]
@tmorehouse What's the problem about the index SCSS file? Wouldn't it make the compilation process much easier?

**Comment by tmorehouse** [comment]
The other reason for keeping the component SCSS/CSS in the same dir as the component would be for @pi0 's plan on making components available as sub-modules (part of the reason why each component has it's own package.json file)

**Comment by tmorehouse** [comment]
@jackmu95 sure!

I just created SCSS versions of all the CSS files (CSS is still there as is, as imported).  The SCSS fiels are just sitting there.

In /scripts/build we could add (or create a script entry in package.json):

```sh
  # bundle all the scss files together
  bundle-scss -m "../src/components/*/style.scss" -d "dist/scss/bootstrap-vue.css"
  # create a convenience file of all the variables for reference
  bundle-scss -m "../src/components/*/_variables.scss" -d "dist/scss/variables.scss"
```

And then the final `bootstrap-vue.scss` could be compiled into `bootstrap-vue.css` (using bootstrap default variable values by specifying the three required includes, maybe using gulp-sass to do all the grunt work, or whatever you may think would be good.

The `variables.scss` file would just be a convenience file for users to reference (as it would already be included in the `booststrap-vue.scss` file).

**Comment by tmorehouse** [comment]
Oops, hit the wrong button :p
